### PR TITLE
Migrate from dep to Golang 1.11 Modules

### PR DESCRIPTION
**Pull request by ChefAustin**
[ Going off this conversation: https://macadmins.slack.com/archives/CCDDKMQ8K/p1553880610004400 ]

This PR will deprecate the use of `deps` in lieu of Modules introduced in Golang 1.11. It also amends the `Makefile` to remove the `go get` statements` as well as introduces `gomodcheck` (thanks to @groob).

I'm only about 60% certain that this is entirely correct, but figured I'd take a whack at it anyway. Happy Friday.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,10 +62,11 @@ endef
...
 
-deps:
-	go get -u github.com/golang/dep/...
-	go get -u github.com/golang/lint/golint
-	dep ensure -vendor-only -v
+gomodcheck:
+	@go help mod > /dev/null || (@echo gorilla requires Go version 1.11 or higher && exit 1)
+
+deps: gomodcheck
+	@go mod download
```
Comment (1dustindavis) [inline Makefile]: I don't think `go mod download` is necessary. Dependencies will be retrieved any time `go build` or `go test` is ran. `deps` can likely be removed all together.

Instead, maybe we should call `gomodcheck` as the first step of `make build` and `make test`, so we exit if `go mod` is not supported.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module github.com/1dustindavis/gorilla
```
Comment (1dustindavis) [inline go.mod]: This file is missing some indirect dependencies. If you run `go get -u` and then `go mod tidy`, it should get updated with all of the latest dependencies,

**Comment by ChefAustin** [comment]
>go mod removes our need for the vendor directory. Let's remove that directory.
Commit 6bf0c25 fixes this.

>There is no go.sum. Running go mod tidy should create one.
Commit 245ae0f checks this one off.

>Add export GO111MODULE=on to the make file so mod won't fail by default.
Done and done with commit 67ac586.

File: Makefile (hunk 1 above)
Comment (ChefAustin) [reply Makefile]: Not positive if I did this as requested, but in commit 6c9b9b3 I threw a `gomodcheck` into the `prebuild` command... directive?... block?... 🤷‍♂️ And removed altogether the `deps:` command... directive?... block?...

File: go.mod (hunk 2 above)
Comment (ChefAustin) [reply go.mod]: Commit 2727219 to the rescue.

**Comment by 1dustindavis** [comment]
It looks like AppVeyor is failing because of [this bug](https://github.com/golang/go/issues/30446) that affects Go 1.11.3 or earlier. They have Go 1.12 planned for the Appveyor images soon ("2-4 weeks" on March 5th).

**Comment by 1dustindavis** [comment]
The changes look good to me and I was able to successfully lint, build, and test on my machine.

Thanks again @ChefAustin!!

**Pull request merged**